### PR TITLE
feat: Update Android Target SDK to 30

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **feat**

This PR resolves [CX-2126]

### Description
- Update Android SDK to 30

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Update android target SDK to 30 - mounir

</details>

[CX-2126]: https://artsyproduct.atlassian.net/browse/CX-2126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Update android target SDK to 30 - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6e6127e809935aa80532eb338b6e4c601783d0b0
</p>

**Comment by MounirDhahri** [comment]
I created a build and tested it for a while and everything seems to be working fine.

File: android/app/build.gradle [hunk 1]
```diff
@@ -204,6 +204,13 @@ android {
 }
 
 dependencies {
+    // This is required to fix an issue that causes the app to crash on startup
+    // This can be removed as soon as we get to a react-native version above 0.66
+    // See https://github.com/facebook/react-native/issues/31725#issuecomment-901682370
+    api(platform("com.squareup.okhttp3:okhttp-bom:4.9.1"))
+    api("com.squareup.okhttp3:okhttp")
+    api("com.squareup.okhttp3:logging-interceptor")
+
```
Comment (MounirDhahri) [inline android/app/build.gradle:213]: See https://github.com/facebook/react-native/issues/31725#issuecomment-901682370 for more details

Comment (brainbicycle) [reply android/app/build.gradle:213]: oh man this just reiterates we have to get to a place where we can upgrade react-native, thanks for fixing

**Comment by brainbicycle** [comment]
Thank you!!

File: android/app/build.gradle (hunk 1 above)
Comment (MounirDhahri) [reply android/app/build.gradle:213]: I absolutely agree!

**Pull request merged**